src/Ledger/Foreign/HSLedger.agda

```diff
@@ -91,56 +97,58 @@ HSP2ScriptStructure .validPlutusScript?    = λ ()
 HSScriptStructure : ScriptStructure ℕ ℕ ℕ
 HSScriptStructure = record { p1s = HSP1ScriptStructure ; ps = HSP2ScriptStructure }
 
-open import Data.Nat
-open import Data.Nat.Properties using (+-0-commutativeMonoid; _≟_)
-open import Ledger.TokenAlgebra
 open import Ledger.Transaction
+import Ledger.TokenAlgebra as TA
+open import Data.Nat.Properties using (+-0-commutativeMonoid; _≟_)
+open import Data.Nat
+
+module _ where
 
-coinTokenAlgebra : TokenAlgebra
-coinTokenAlgebra = record
-  { PolicyId                = ℕ
-  ; Value-CommutativeMonoid = +-0-commutativeMonoid
-  ; coin                    = id
-  ; inject                  = id
-  ; policies                = λ x → ∅
-  ; size                    = λ x → 1 -- there is only ada in this token algebra
-  ; property                = λ x → refl
-  ; coinIsMonoidMorphism    = record
-    { mn-homo = record
-      { sm-homo = record { ⟦⟧-cong = λ z → z ; ∙-homo  = λ x y → refl }
-      ; ε-homo  = refl
+  open TA ℕ
+  open TA.TokenAlgebra
+  coinTokenAlgebra : TokenAlgebra
+  Value-CommutativeMonoid coinTokenAlgebra    = +-0-commutativeMonoid
```

Comment:
I find this very difficult to read since the copattern is separated from its right hand side. Can you either put the copattern on the right (i.e. `coinTokenAlgebra .coin = ...`) or use `record` syntax?